### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by aknysh**
## what
* Initial implementation

## why
* Move all atmos-related functionality from `terraform-provider-utils` to `terraform-provider-atmos` (leaving `terraform-provider-utils` with generic functionality like deep-merging and yaml/json utils)
* Add initial implementation of `atmos_label` data source (replacement of `terraform-null-label` with more functionality)

## test

```
locals {
  namespace   = "eg"
  tenant      = "plat"
  environment = "ue2"
  stage       = "dev"
  name        = "test"
  delimiter   = "-"

  result1 = yamldecode(data.atmos_label.example1.output)
}

data "atmos_label" "example1" {
  namespace   = local.namespace
  tenant      = local.tenant
  environment = local.environment
  stage       = local.stage
  name        = local.name
  delimiter   = local.delimiter
}
```

```
terraform init
terraform apply
```

```
data.atmos_label.example1: Reading...
data.atmos_label.example1: Read complete after 0s [id=513f0dba65346a6b1fc0f4b91032d0df763159e6]

Changes to Outputs:
  + label1 = "eg-plat-ue2-dev-test"

You can apply this plan to save these new output values to the Terraform state, without changing any real infrastructure.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes


Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

label1 = "eg-plat-ue2-dev-test"
```

### CODE REVIEW

File: templates/index.md.tmpl [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+layout: ""
+page_title: "Provider: Cloud Posse Utils"
+description: |-
+  The utils provider contains data sources to perform tasks that are difficult to perform in raw Terraform.
+---
+
+# Cloud Posse Utils Provider
+
+The utils provider contains data sources to perform tasks that are difficult to perform in raw Terraform.
```
Comment (Nuru) [inline templates/index.md.tmpl]: Update for Atmos provider

File: .go-version [hunk 2]
```diff
@@ -0,0 +1 @@
+1.16.3
```
Comment (Nuru) [inline .go-version]: Can we bump this to the current [1.18](https://tip.golang.org/doc/go1.18)?
```suggestion
1.18.3
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+## 1.0.0
```
Comment (Nuru) [inline CHANGELOG.md]: Do you want to start with a 1.0 release? I suggest we start with a 0.1.0 release and release 1.0 once we are sure we have preserved existing functionality.

File: go.mod [hunk 4]
```diff
@@ -0,0 +1,13 @@
+module github.com/cloudposse/terraform-provider-atmos
+
+go 1.16
```
Comment (Nuru) [inline go.mod]: Can we update to `go` 1.18?

**Review by Nuru — CHANGES_REQUESTED** [review]
Let's add [.github/.github-update-disabled](https://github.com/cloudposse/build-harness/blob/6665d583704a4fd2befef64a21a94bca266bda4d/.github/.github-update-disabled) to keep from accidentally adding all the stuff for Terraform modules.


What is the purpose of having both `examples/complete/components/helmfile/echo-server/helmfile.yaml` and `examples/complete/components/helmfile/infra/infra-server/helmfile.yaml`? 

The examples are all out of date, which is not that important, but it would be nice to get them updated to using 1.x or 2.x versions of modules where we have them. Not saying we need to do that in this release, but something I'd like to see before we release v1.0 of this provider.

Can we use the new [Terraform Plugin Framework](https://www.terraform.io/plugin/framework) instead of the v2 SDK? I am particularly interested in its separate support for Data Sources (not folded in with resources) and the ability to tell the difference between null, unknown, and empty values. See [Which SDK Should I Use?](https://www.terraform.io/plugin/which-sdk)

File: .go-version (hunk 2 above)
Comment (aknysh) [reply .go-version]: the old provider framework had issues with G0 1.17.
that's why we are at 1.16

File: CHANGELOG.md (hunk 3 above)
Comment (aknysh) [reply CHANGELOG.md]: all exisiting functionality from `utils-provider` is there, including all the examples and tests

File: templates/index.md.tmpl (hunk 1 above)
Comment (aknysh) [reply templates/index.md.tmpl]: updated this

File: CHANGELOG.md (hunk 3 above)
Comment (aknysh) [reply CHANGELOG.md]: but if you want to look at the new Provider Framework, it's a completely diff story

Comment (aknysh) [reply CHANGELOG.md]: but let's start with `0.1.0` in any case, updated

**Comment by mcalhoun** [comment]
Closing as outdated since we aren't going down this route (and it keeps showing as an open PR I need to review).

**Pull request closed without merging**